src/IdealizedSimulations/neverworld_simulation.jl

```diff
@@ -0,0 +1,409 @@
+    scotia_ridge_center_latitude :: Float64
+end
+
+function NeverworldBathymetry(grid;
+                              abyssal_depth = 4000,
+                              shelf_depth = 200,
+                              shelf_width = 2.5,
+                              rim_width = shelf_width / 8,
+                              slope_width = shelf_width,
+                              southern_channel_boundary = -60,
+                              northern_channel_boundary = -40,
+                              scotia_ridge_height = 2000,
+                              scotia_ridge_radius = 10,
+                              scotia_ridge_width = 2,
+                              scotia_ridge_center_longitude = 0,
+                              scotia_ridge_center_latitude = (southern_channel_boundary + northern_channel_boundary) / 2)
+
+    # Use grid spacing for "beach width"
+    Δ = max(grid.Δλᶠᵃᵃ, grid.Δφᵃᶠᵃ)
+
+    # Construct a cubic spline of the form `basin_depth(r)`, representing
+    # the "basin component" of the Neverworld bathymetry where `r` is the distance
+    # to the edge of the Neverworld (with units of degrees).
+    r_coast = Δ
+    r_beach = Δ + rim_width
+    r_mid_shelf = Δ + rim_width + shelf_width / 2
+    r_shelf = Δ + rim_width + shelf_width
+    r_abyss = Δ + rim_width + shelf_width + slope_width
+
+    Nx, Ny, Nz = size(grid)
+    x_max = xnode(Nx+1, 1, 1, grid, f, c, c) - xnode(1, 1, 1, grid, f, c, c)
+    y_max = ynode(1, Ny+1, 1, grid, c, f, c) - ynode(1, 1, 1, grid, c, f, c)
+    r_max = max(x_max, y_max)
+
+    basin_rim_distances = [0, r_coast,     r_beach,  r_mid_shelf,     r_shelf,       r_abyss,         r_max]
+    basin_depths        = [0, 0,       shelf_depth,  shelf_depth, shelf_depth, abyssal_depth, abyssal_depth]
+    coastline_spline = CubicSplineFunction{:x}(basin_rim_distances, basin_depths)
+
+    R = scotia_ridge_radius 
+    w = scotia_ridge_width
+    H = abyssal_depth
+    h = H - scotia_ridge_height
+
+    # The so-called "clipped cone"
+    scotia_ridge(r) = max(h, H * min(1, abs(r - R) / w))
+
+    return NeverworldBathymetry(grid,
+                                coastline_spline,
+                                scotia_ridge,
+                                Float64(rim_width),
+                                Float64(slope_width),
+                                Float64(shelf_width),
+                                Float64(shelf_depth),
+                                Float64(abyssal_depth),
+                                Float64(southern_channel_boundary),
+                                Float64(northern_channel_boundary),
+                                Float64(scotia_ridge_height),
+                                Float64(scotia_ridge_radius),
+                                Float64(scotia_ridge_width),
+                                Float64(scotia_ridge_center_longitude),
+                                Float64(scotia_ridge_center_latitude))
```

Comment:
why all these `Float64` here instead of `eltype(grid)`?

Reply:
these functions are not called on the fly. They are called just once when `ImmersedBoundaryGrid` is constructed. The bathymetry data is always eltype(grid). I put `Float64` for convenience, and also not to mislead people into thinking that it actually mattered what datatype was used. We could also leave the datatype unspecified, maybe at some cost to performance during setup.